packages/@react-aria/collections/src/CollectionBuilder.tsx

```diff
@@ -114,14 +113,6 @@ function useCollectionDocument<T extends object, C extends BaseCollection<T>>(cr
     return document.getCollection();
   }, [document]);
   let collection = useSyncExternalStore(subscribe, getSnapshot, getServerSnapshot);
-  useLayoutEffect(() => {
```

Comment:
Bug found on the illustrations page when wrapping a collection in Suspense: when switching illustration types, the content would not update in the collection. This was because during Suspense React "unmounts" effects by calling their cleanup functions, but doesn't actually remove the component from the DOM. Before it re-mounts the effect, it removes the old content from the collection. We had this optimization where we ignored removals when unmounted, so these would never be processed. I tested in Storybook with our large collection stories and didn't notice a performance issue without this, so perhaps we fixed the original issue somewhere else in the meantime.